src/python/WMCore/Database/MongoDB.py

```diff
@@ -5,11 +5,15 @@
 
 # futures
 from __future__ import division, print_function
-
 from builtins import str, object
 
+try:
+    import mongomock
+except ImportError:
+    # this library should only be required by unit tests
+    mongomock = None
```

Comment:
That is a good workaround, but if we go that way we should then walk to the end. 

We do know that it is supposed to be used only in the unit tests. But, since the class itself provides the option for an object to be created with `mockMongoDB=True`, then  if someone, who is missing the `mongomock` library, does try to benefit from the flag, this would generate the following error on line [53](https://github.com/dmwm/WMCore/blob/4d64d4aab684a7a18b852fba5ab52dba7479e2d3/src/python/WMCore/Database/MongoDB.py#L53) :

```
In [1]: mongomock = None

In [2]: client = mongomock.MongoClient()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-233abe3373bf> in <module>
----> 1 client = mongomock.MongoClient()

AttributeError: 'NoneType' object has no attribute 'MongoClient'
```

In this case I would suggest to add some extra protection on line in question, like:
```
In [7]: if mockMongoDB and mongomock is not None:
   ...:     self.client = mongomock.MongoClient()

```

Reply:
I actually wanted it to crash indeed, since it would be a setup that people should never even try. Nothing as a big error in our face :)
But I think I can do something slightly better here, still along the same line of thought. New change coming up in a few minutes.